src/components/ResourceLink.tsx

```diff
@@ -0,0 +1,60 @@
+import { Icon } from "@canonical/react-components";
+import { FC } from "react";
+import { useNavigate } from "react-router-dom";
+
+export type ResourceType =
+  | "container"
+  | "virtual-machine"
+  | "snapshot"
+  | "profile"
+  | "project"
+  | "cluster-member"
+  | "network"
+  | "pool"
+  | "volume"
+  | "iso-volume"
+  | "image"
+  | "oidc-identity"
+  | "certificate"
+  | "auth-group"
+  | "device";
+
+const resourceIconMap: Record<ResourceType, string> = {
+  container: "lxd-container",
+  "virtual-machine": "pods",
+  snapshot: "snapshot",
+  profile: "repository",
+  project: "folder",
+  "cluster-member": "single-host",
+  network: "exposed",
+  pool: "",
+  volume: "",
+  "iso-volume": "iso",
+  image: "image",
+  "oidc-identity": "user",
+  certificate: "certificate",
+  "auth-group": "user-group",
+  device: "units",
+};
+
+interface Props {
+  type: ResourceType;
+  value: string;
+  to: string;
+}
+
+const ResourceLink: FC<Props> = ({ type, value, to }) => {
+  const icon = resourceIconMap[type];
+  const navigate = useNavigate();
+  return (
+    <button className="p-chip is-inline" onClick={() => navigate(to)}>
```

Comment:
I think button is the wrong component to use and this should be a `Link` to be semantically correct. Then the browser would display the target url correctly and you can do things like open in a new tab.